src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -179,56 +164,66 @@ internal static SqlConnection GetSqlConnection(ConnectionInfo connInfo)
             return null;
         }
 
-        internal void InitializeBackup(CDataContainer dataContainer, SqlConnection sqlConnection)
+        internal BackupConfigInfo GetBackupConfigInfo(CDataContainer dataContainer, SqlConnection sqlConnection, string databaseName)
         {
-            this.backupUtilities.Initialize(dataContainer, sqlConnection);
-        }
-
-        internal void SetBackupInput(BackupInfo input)
-        {   
-            this.backupUtilities.SetBackupInput(input);
+            BackupOperation backupOperation = new BackupOperation();
+            backupOperation.Initialize(dataContainer, sqlConnection);
+            return backupOperation.GetBackupConfigInfo(databaseName);
         }
 
-        internal void PerformBackup()
+        internal BackupOperation SetBackupInput(CDataContainer dataContainer, SqlConnection sqlConnection, BackupInfo input)
         {
-            this.backupUtilities.PerformBackup();
+            BackupOperation backupOperation = new BackupOperation();
+            backupOperation.Initialize(dataContainer, sqlConnection);
+            backupOperation.SetBackupInput(input);
+            return backupOperation;
         }
-
-        internal BackupConfigInfo GetBackupConfigInfo(string databaseName)
+                
+        internal void PerformBackup(BackupOperation backupOperation)
         {
-            return this.backupUtilities.GetBackupConfigInfo(databaseName);
+            backupOperation.PerformBackup();
         }
-
+        
         /// <summary>
         /// Create a backup task for execution and cancellation
         /// </summary>
         /// <param name="sqlTask"></param>
         /// <returns></returns>
-        internal async Task<TaskResult> BackupTask(SqlTask sqlTask)
+        internal async Task<TaskResult> BackupTaskAsync(SqlTask sqlTask)
         {
             sqlTask.AddMessage(SR.Task_InProgress, SqlTaskStatus.InProgress, true);
-            Task<TaskResult> performTask = this.PerformTask();
-            Task<TaskResult> cancelTask = this.CancelTask(sqlTask);
+
+            IBackupOperation backupOperation = sqlTask.TaskMetadata.Data as IBackupOperation;
+            AutoResetEvent backupCompletedEvent = new AutoResetEvent(initialState: false);
+            Task<TaskResult> performTask = this.PerformTaskAsync(backupOperation);
+            Task<TaskResult> cancelTask = this.CancelTaskAsync(backupOperation, sqlTask.TokenSource.Token, backupCompletedEvent);
```

Comment:
nit: we're typically not using 'this.' prefix when calling methods..only when access fields or properties per typical .Net code guidelines.